src/Head/index.js

```diff
@@ -0,0 +1,13 @@
+/* @flow */
+
+export type _Head =
+    & (([]) => void)
+    & (<T>([T]) => T)
+    & (<T, A>([T, A]) => T)
+    & (<T, A, B>([T, A, B]) => T)
+    & (<T, A, B, C>([T, A, B, C]) => T)
+    & (<T, A, B, C, D>([T, A, B, C, D]) => T)
+    & (<T, A, B, C, D, E>([T, A, B, C, D, E]) => T)
+    & (<T, V: $ReadOnlyArray<T>>(arr: V) => T)
```

Comment:
@Ktan32 do we have any restrictions to do something like this?

```js
type _Head = (
	& (([]) => void)
	& (<T>(T) => $ElementType<T, 0>)
)
```

I've checked the tests, so it works the same way [link](https://flow.org/try/#0C4TwDgpgBA+gEhAhgEygXigCgFAEgBkWmA2gLoCU6AfFAG4D2AlsuXoZgDwAqVmXlaGgBIAogBsIAWwgA7YF3ARuAGigAGKq1bZQkKAhTcaGIQGFEYsR3hJkqntmwB6AFQvsUF1C4BXMBIBnDxcnbABvDywfGWQIADNGGQhkAC59Ww4yTUjXdyhPKFEAD0gAY2ARACdK+krIkMiAbkwZH0s0g2RM0k0oJycoCGrax3zmgEYOjOJxntZ83PrCkRKIcqqauvyGscwAJinDGbm+gaHNxyb9w67iPdVZ7IW3JeKyiuGtz1DdyfSj+5QR6UfqDT6XXYAZhumUhqkBwJyL22y1W63B2x+UGaB3+tzhUARJ1B5xGo2xmAALDDiJTVASiU8+siCm81h8LpirtC8Zk6VAGQ9iWdweTmgBWGni1T8wVAuZIvKslbvDYjLm7am84jSqCy+FC3ok0W7ABsNNNql1+sJhvmzKVXjZ6M53yuku1lqg1vpBvlmmNFwAvo5QQBBaqIEBBCL5TB-TocVqSABGQyy9pxMOTacqGbF8ZhQgASrYAPIyMQgCOVKNJnypoZUJlZ3kl8uV6uRkD1xuVZusEPOAYAVSSaOAySguggMcipXoMgCwCgjEk-jSiBkIHQUEQ1D3xDUpHni+XUAAFrY0jYULu1-5Fa8Vey1V8dpfbC02mIQSKLk+KLOhy6puvkV4oJgADkiCIFBf5ggBzyOqiqoYmBn6QWEQYIaSdSAcqE5vvUWIQcgmDAJUPgQLhJoOs+RHoR+ZGYAAyiAqb0GI0FkfBtHBkAA)

Reply:
and I'm not sure that head of empty tuple should be safety

Reply:
Completely true about using `$ElementType`. In my previous attempt I had problem with inappropriate arguments (like object or function) which did result in a `void`, not an `error`.

Yet it may be fixed this way:
```js
    & (<T>(T: $ReadOnlyArray<mixed>) => $ElementType<T, 0>)
```
My fault, I'll fix this.

Reply:
>and I'm not sure that head of empty tuple should be safety
Why not? It seems to be a legal operation
Plus, `$ElementType<[], 0>` won't raise Error and will return `empty`